src/main/kotlin/de/simpletactics/wiki/lib/adapter/persistence/WikiSqlAdapter.kt

```diff
@@ -0,0 +1,102 @@
+package de.simpletactics.wiki.lib.adapter.persistence
+
+import de.simpletactics.wiki.lib.adapter.dto.EntryEntity
+import de.simpletactics.wiki.lib.adapter.dto.TopicEntity
+import de.simpletactics.wiki.lib.adapter.persistence.mapper.EntryMapper
+import de.simpletactics.wiki.lib.adapter.persistence.mapper.IdMapper
+import de.simpletactics.wiki.lib.adapter.persistence.mapper.TopicRowMapper
+import de.simpletactics.wiki.lib.model.WikiNotFoundException
+import de.simpletactics.wiki.lib.model.WikiType
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.stereotype.Component
+
+@Component
+class WikiSqlAdapter(
+    private val jdbc: JdbcTemplate,
+) {
+
+    fun getWikiType(id: Int): WikiType? {
+        val result = jdbc.queryForList("SELECT type FROM wiki WHERE id = ?", id)
+        return if (result.size == 1 && result.first().containsKey("type")) {
+            val typeAsString = result.first()["type"].toString()
+            WikiType.valueOf(typeAsString)
+        } else {
+            null
+        }
+    }
```

Comment:
Weiß nicht was besser ist:
fun getWikiType(id: Int): WikiType? {
    val result = jdbc.queryForList("SELECT type FROM wiki WHERE id = ?", id)

    check(result.size == 1 && result.first().containsKey("type")) {
        "Invalid result for id $id: Expected exactly one row with a 'type' column"
    }

    return WikiType.valueOf(result.first()["type"].toString())
}

Hier würden wir noch einen Fehler werfen.

Reply:
Können wir auch mehr als 1 zurückbekommen? Fall nicht könnte man das glaube ich auch anders schreiben. Ich versuche mittlerweile else so gut es geht zu vermeiden.